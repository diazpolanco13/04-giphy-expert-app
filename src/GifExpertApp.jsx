import React, { useState } from "react";
import AppCategory from "./Components/AppCategory";
import GifGrid from "./Components/GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Casa"]);

  return (
    <div>
      <h2 className="text-3xl pb-4 font-bold" >GifsApp</h2>
      <AppCategory setCategories={setCategories} categories={categories} />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </div>
  );
};

export default GifExpertApp;
