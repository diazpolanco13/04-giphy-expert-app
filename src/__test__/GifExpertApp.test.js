import React from "react";
import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Probando el componente", () => {
  let wrapper = shallow(<GifExpertApp />);
  beforeEach(() => {
    //Reinicia el componente con dada prueba
    wrapper = shallow(<GifExpertApp />);
  });

  test("1- Debe mostrar <ConunterApp/> correctamente ", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
