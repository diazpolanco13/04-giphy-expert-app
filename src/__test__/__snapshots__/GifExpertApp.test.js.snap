// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente 1- Debe mostrar <ConunterApp/> correctamente  1`] = `
<div>
  <h2>
    GitExpertApp
  </h2>
  <AppCategory
    categories={
      Array [
        "Dragon Ball",
      ]
    }
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
